src/Tools/dotnet-watch/src/MSBuildEvaluationFilter.cs

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.DotNet.Watcher.Tools
+{
+    public class MSBuildEvaluationFilter : IWatchFilter
+    {
+        // File types that require an MSBuild re-evaluation
+        private static readonly string[] _msBuildFileExtensions = new[]
+        {
+            ".csproj", ".props", ".targets", ".fsproj", ".vbproj", ".vcxproj",
```

Comment:
A little worried about forgetting files here. Like what about global.json?

Reply:
Yeah, would it be better to check only for non-msbuild file types, like *.cs, .razor, etc.?

Reply:
The current watcher by default does not include arbitrary files include global.json. While it could be possible that you could add these files to the watch list, it might be good to understand if these are part of the regular dev workflow. Specifically with global.json, changing the sdk would also require you to restart dotnet watch to get the corresponding global tool \ runtime etc.

Reply:
How do you feel about having a way to opt out of this behavior? That way if we do something terribly incorrect, users would have to mitigate this until we can fix it.

Reply:
Yeah that may be a good idea.

Reply:
Done